client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -511,113 +515,64 @@ def inject_ayon_environment(deadlinePlugin):
+            )
 
-        for env, val in environment.items():
-            # Add the env var for the Render Plugin that is about to render
-            deadlinePlugin.SetEnvironmentVariable(env, val)
-            # Add the env var for current calls to `DeadlinePlugin.RunProcess`
-            deadlinePlugin.SetProcessEnvironmentVariable(env, val)
+        output_dir = _get_output_dir(job)
+
+        shared_env_group = os.environ.get("AYON_SITE_ID")
+        # drive caching of environment variables with env var
+        # it is recommended to use same value AYON_SITE_ID for 'same'
+        # render nodes (eg. same OS etc.)
+        if shared_env_group:
+            print("Caching of environment file will be used.")
+            environment_file_name = f"env_{shared_env_group}.json"
+            export_dir_url = os.path.join(
+                output_dir,
+                ".ayon_env_cache"
+            )
 
-        args_str = subprocess.list2cmdline(args)
-        print(">>> Executing: {} {}".format(exe, args_str))
-        process_exitcode = deadlinePlugin.RunProcess(
-            exe, args_str, os.path.dirname(exe), -1
-        )
+            if not os.path.exists(export_dir_url):
+                os.makedirs(export_dir_url, exist_ok=True)
 
-        if process_exitcode != 0:
-            print(
-                "Failed to run AYON process to extract environments. Trying"
-                " to use legacy arguments."
-            )
-            legacy_args_str = subprocess.list2cmdline(legacy_args)
-            process_exitcode = deadlinePlugin.RunProcess(
-                exe, legacy_args_str, os.path.dirname(exe), -1
+            export_url = os.path.join(
+                export_dir_url,
+                environment_file_name)
+            _wait_for_in_progress(job, export_url)
+        else:
+            # no caching - existing behavior
+            temp_file_name = "{}_{}.json".format(
+                datetime.utcnow().strftime("%Y%m%d%H%M%S%f"),
+                str(uuid.uuid1())
             )
-            if process_exitcode != 0:
-                raise RuntimeError(
-                    "Failed to run AYON process to extract environments."
+            export_url = os.path.join(tempfile.gettempdir(), temp_file_name)
+
+        if not os.path.exists(export_url):
+            print(f"'{export_url}' doesn't exist yet, extracting...")
+            temp_export_url = f"{export_url}.tmp"
+            with open(temp_export_url, "w"):
+                pass
+            try:
+                _extract_environments(
+                    ayon_server_url,
+                    ayon_api_key,
+                    ayon_bundle_name,
+                    deadlinePlugin,
+                    exe,
+                    temp_export_url,
+                    job
                 )
+                if not os.path.exists(export_url):
+                    print(f"Creating env var file {export_url}")
+                    shutil.move(temp_export_url, export_url)
```

Comment:
If it's on same disk (which I think always is) you can use faster `os.rename`.
```suggestion
                    os.rename(temp_export_url, export_url)
```